bigframes/clients.py

```diff
@@ -162,3 +164,25 @@ def _get_service_account_if_connection_exists(
             pass
 
         return service_account
+
+
+def get_connection_name_full(
+    connection_name: Optional[str], default_project: str, default_location: str
```

Comment:
awkward to have optional argument followed by mandatory arguments, would be nice to have it in the end

Reply:
Optional only means it can be value None. It is OK to put it at the front, as it is the most important param of the function.

Not really "optional", which means it has a default value. Then it has to be at the end.